### PR TITLE
chore:  Add initial unit & e2e tests

### PR DESCRIPTION
**Pull request by grantnorwood**
* Add initial unit & e2e tests.
* Add npm script for test coverage, and ignore /coverage in git.
* Change from using happy-dom to jsdom in vitest.
* Moved components into subfolders so unit tests can be colocated.
* Ignore /coverage in Prettier config.
* Created [Cypress Dashboard](https://cloud.cypress.io/projects/6s6892/runs) account _(free plan with 3 users)_
* Added Cypress e2e tests on commits to PR and to main _(runs pretty fast on commit – about 2 mins – but perhaps we can execute these on automated pushes to staging once that's setup ...)_
  * Created `CYPRESS_PROJECT_ID` & `CYPRESS_RECORD_KEY` secrets in the repo
  * Created `cypress.yml` GHA workflow to be called by other workflows

# TODO
* [x] We need to update `pull-request.yml` to use the `development.yml@main` reference **before** merging.

# Known Issues
* Getting an error recording to Cypress Cloud:  
  * The two relevant `CYPRESS_` env variables are empty in the [Action logs](https://github.com/remix-austin/remixaustin-com/actions/runs/3555158100/jobs/5971684643#step:9:11) for some reason ...
    ```
    env:
      npm_config_cache: /home/runner/.npm
      PORT: [8](https://github.com/remix-austin/remixaustin-com/actions/runs/3555158100/jobs/5971684643#step:9:8)811
      CYPRESS_PROJECT_ID: 
      CYPRESS_RECORD_KEY: 
      GITHUB_TOKEN: ***
      DEBUG: cypress:*
    ```
  * GitHub integration is setup for the [Cypress Dashboard Project](https://cloud.cypress.io/projects/6s6892/runs)
  * Secrets for `CYPRESS_PROJECT_ID` & `CYPRESS_RECORD_KEY` were added to the repo
  * Not sure why this is happening ... maybe because I configured the Cypress Cloud project with the `main` branch as default?  I can't seem to unset that, though.
  * **This is not a blocker for merge**, but something that may require further troubleshooting.  The Cypress headless tests are still run successfully!

### CODE REVIEW

**Review by KarenDouglas — COMMENTED** [review]
I am not well versed in testing software to be able to give meaningful approval or disapproval, at this time.  I am learning about vitest, cypress and testing library, though,  so I look forward to more opportunities to review code in the future.

File: app/components/Navbar/Navbar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      </Router>
+    );
+
+    expect(screen.getByRole("navigation")).toBeInTheDocument();
+
+    expect(
+      screen.getByRole("link", { name: homepageLinkTitle })
+    ).toBeInTheDocument();
+
+    expect(screen.getAllByRole("link", { name: "Meetup" })).toHaveLength(2);
```
Comment (tcschiller) [inline app/components/Navbar/Navbar.test.tsx:21]: Length of 2 because there's a mobile and a desktop link?

File: app/utils.test.ts [hunk 2]
```diff
@@ -1,13 +1,71 @@
...
+      expect(validateEmail("not-an-email")).toBe(false);
+      expect(validateEmail("n@")).toBe(false);
+    });
+
+    test("validateEmail returns true for emails", () => {
+      expect(validateEmail("kody@example.com")).toBe(true);
+    });
+  });
+
+  describe("safeRedirect", () => {
```
Comment (tcschiller) [inline app/utils.test.ts:18]: This is fantastic.

File: app/utils.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ export function safeRedirect(
   to: FormDataEntryValue | string | null | undefined,
   defaultRedirect: string = DEFAULT_REDIRECT
 ) {
-  if (!to || typeof to !== "string") {
+  if (!to || typeof to !== "string" || to === "") {
```
Comment (tcschiller) [inline app/utils.ts:17]: Weird, I would have thought the first conditional in this if statement would suffice for empty strings.

**Review by tcschiller — APPROVED** [review]
Overall looks great. Some minor comments.

**Comment by tcschiller** [comment]
Ack, saw your Cypress Cloud troubles comment just now. @grantnorwood are you still getting that, by chance? Or did adding the project ID help?

**Comment by grantnorwood** [comment]
> I am not well versed in testing software to be able to give meaningful approval or disapproval, at this time. I am learning about vitest, cypress and testing library, though, so I look forward to more opportunities to review code in the future.

@KarenDouglas No problem at all, reviewing others' code is a great opportunity for learning!  And now you have a project you can practice with 😉

File: app/components/Navbar/Navbar.test.tsx (hunk 1 above)
Comment (grantnorwood) [reply app/components/Navbar/Navbar.test.tsx:21]: Correct!

File: app/utils.test.ts (hunk 2 above)
Comment (grantnorwood) [reply app/utils.test.ts:18]: 🙏

File: app/utils.ts (hunk 3 above)
Comment (grantnorwood) [reply app/utils.ts:17]: Not weird at all, this is what you get from a manager who doesn't code everyday 😉  I'll remove that empty string conditional.

**Comment by grantnorwood** [comment]
> Ack, saw your Cypress Cloud troubles comment just now. @grantnorwood are you still getting that, by chance? Or did adding the project ID help?

@tcschiller  Yes, the error message seems to think this is being run from a "forked PR", though being in the same repo I would expect it work _(or would just reference a "PR")_.

```
Warning: It looks like you are trying to record this run from a forked PR.

The Record Key is missing. Your CI provider is likely not passing private environment variables to builds from forks.

These results will not be recorded.

This error will not alter the exit code.
```

I'll be curious to see if it runs successfully once merged to `main`, but I still like the idea of it running from PRs.  The error may still be correct in that GitHub isn't passing the secrets over correctly, but the secrets are created for the entire repo, so 🤷

**Comment by grantnorwood** [comment]
Fixes issue https://github.com/remix-austin/remixaustin-com/issues/2

**Pull request merged**